app/screens/AppNavigator/screens/Dex/PoolSwap/PoolSwapScreen.tsx

```diff
@@ -0,0 +1,209 @@
+import { AddressToken } from '@defichain/whale-api-client/dist/api/address'
+import { PoolPairData } from '@defichain/whale-api-client/dist/api/poolpair'
+import { MaterialIcons } from '@expo/vector-icons'
+import { StackScreenProps } from '@react-navigation/stack'
+import BigNumber from 'bignumber.js'
+import React, { useEffect, useState } from 'react'
+import { Control, Controller, useForm } from 'react-hook-form'
+import { ScrollView, TouchableOpacity, View } from 'react-native'
+import NumberFormat from 'react-number-format'
+import tailwind from 'tailwind-rn'
+import { Text, TextInput } from '../../../../../components'
+import { getTokenIcon } from '../../../../../components/icons/tokens/_index'
+import { PrimaryButton } from '../../../../../components/PrimaryButton'
+import { PrimaryColor, PrimaryColorStyle } from '../../../../../constants/Theme'
+import { useTokensAPI } from '../../../../../hooks/wallet/TokensAPI'
+import { translate } from '../../../../../translations'
+import LoadingScreen from '../../../../LoadingNavigator/LoadingScreen'
+import { DexParamList } from '../DexNavigator'
+
+interface TokenForm {
+  control: Control
+  controlName: string
+  token: AddressToken
+  onMaxPress?: (amount: string) => void
+  title: string
+}
+
+interface SwapSummaryItems {
+  poolpair: PoolPairData
+  tokenA: AddressToken
+  tokenB: AddressToken
+}
+
+type Props = StackScreenProps<DexParamList, 'PoolSwapScreen'>
+
+export function PoolSwapScreen ({ route }: Props): JSX.Element {
+  const poolpair = route.params.poolpair
+  const tokens = useTokensAPI()
+  const [tokenAForm, tokenBForm] = ['tokenA', 'tokenB']
+  const [tokenA, setTokenA] = useState<AddressToken>()
+  const [tokenB, setTokenB] = useState<AddressToken>()
+  const { control, setValue, formState: { isValid }, getValues, trigger } = useForm({ mode: 'onChange' })
+
+  function onSubmit (): void {
+    console.log(isValid)
+    console.log(getValues())
+  }
+
+  function swapToken (): void {
+    setTokenA(tokenB)
+    setTokenB(tokenA)
```

Comment:
curious this doesn't cause race condition?

Reply:
It doesn't due to React batching state updates. You can check this for reference - https://github.com/facebook/react/issues/11527#issuecomment-360199710